### DISCUSSION TITLE
How to remove/rename span attributes?

### DISCUSSION THREAD

**Discussion by kroehre** [open-telemetry/opentelemetry-ruby#1335] [Q&A] [answered]
Hi, I'm trying to figure out how to rename the [GraphQL instrumentation span attributes](https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/main/instrumentation/graphql/lib/opentelemetry/instrumentation/graphql/tracers/graphql_tracer.rb#L77-L79) to [semantic names](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/graphql.md) using a span processor. I see how I can add a new attribute, but I don't see how to remove or rename an existing attribute on the span. Thanks!

**Comment by arielvalentin** [comment, marked as answer]
The best place to do this is in the OTel Collector and likely using an attribute processor[^1], however if you are not using a collector you can try to write some custom code for the client SDK. Span attribute processing is not a feature of the client SDK specification so it isn't something we currently support. That may change in the future with the introduction of schema upgrades/transformation. 

Some teams workaround that by injecting a custom span processor and edits the read/write span[^2], however you must insert your custom processor before you export the spans.   

Another option here would be to for you to submit a PR that updates the GraphQL instrumentation with the latest SemConv attributes! It would be a great way for you to get involved and help the project grow :)

[^1]: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/attributesprocessor
[^2]: https://github.com/open-telemetry/opentelemetry-ruby/discussions/1272#discussioncomment-2786455

**Reply by kroehre** [reply]
Thanks for you thoughtful reply @arielvalentin. 
PR: https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/73
